server/src/main/java/org/apache/druid/query/lookup/LookupReferencesManager.java

```diff
@@ -373,11 +374,28 @@ private void takeSnapshot(Map<String, LookupExtractorFactoryContainer> lookupMap
     }
   }
 
-  private void loadAllLookupsAndInitStateRef()
+  /**
+   * Load a set of lookups.
+   * @param lookupsToLoad List of lookup names to load. Pass null to load all lookups.
+   */
+  private void loadLookupsAndInitStateRef(List<String> lookupsToLoad)
   {
     List<LookupBean> lookupBeanList = getLookupsList();
-    if (lookupBeanList != null) {
-      startLookups(lookupBeanList);
+
+    List<LookupBean> lookupBeansToLoad;
+    if (lookupsToLoad != null) {
+      lookupBeansToLoad = new ArrayList<>();
+      if (lookupBeanList != null) {
+        lookupBeansToLoad = lookupBeanList.stream()
+                                          .filter(lookupBean -> lookupsToLoad.contains(lookupBean.getName()))
+                                          .collect(Collectors.toList());
+      }
+    } else {
+      lookupBeansToLoad = lookupBeanList;
+    }
+
+    if (lookupBeansToLoad != null) {
+      startLookups(lookupBeansToLoad);
```

Comment:
This can be simplified:

```suggestion
    final List<String> lookupsToLoad = lookupListeningAnnouncerConfig.getLookupsToLoad();
    final boolean loadAllLookups = lookupsToLoad == null;

    if (lookupBeanList != null) {
      lookupBeanList = lookupBeanList.stream()
                                                            .filter(lookupBean -> loadAllLookups || lookupsToLoad.contains(lookupBean.getName()))
                                                            .collect(Collectors.toList());
      startLookups(lookupBeansToLoad);
```

Reply:
Makes sense, have made the change.